### ISSUE TITLE
Ansible should parse the playbook before asking for sudo or ssh passwords

### ISSUE THREAD

**Issue by ghost** [ansible/ansible#1253]
For example, I have a playbook named remotesetup.yml and I run it from the command line like that:

ansible-playbook remotesetupp.yml -K 

It will ask for the sudo password even if the file doesn't exist (I made a typo, two p at the end).
In the same way, if there's a yaml syntax error in the playbook it will still ask me for the sudo pass even though it can't run the playbook.

Thus, the parsing of the playbook should be done prior to asking for sudo or ssh pass.

**Comment by mpdehaan** [comment]
duplicate of #1227
